src/store/actions/actionCreators.js

```diff
@@ -0,0 +1,101 @@
+import ACTIONS from './actionTypes';
+
+export const getMovie = () => {
+  return (dispatch) => {
+    fetch('http://localhost:4000/movies')
+      .then((res) => res.json())
+      .then((movieData) => {
+        dispatch({ type: `${ACTIONS.GET_MOVIE}`, payload: movieData.data });
+      })
+      .catch((err) => {
+        console.error("Server doesn't response", err);
+      });
+  };
+};
+
+export const filterByGenre = (genre, { allMovies }) => {
+  return (dispatch) => {
```

Comment:
filterByGenre could be a reducer function, your action should deliver only `type` and how it should be filered.

Reply:
will be fixed later